src/containers/Editor/components/views/GraphView/lib/utils/traverse.ts

```diff
@@ -17,242 +12,110 @@ type Traverse = {
+      // or multiple nodes within an object
+      // For simple types we can just read in the child objects value
+      let nodeText = "";
+      if (child.value !== undefined) {
+        nodeText = String(child.value);
+      }
 
-  if (nextType && parentType !== "array" && (nextType === "object" || nextType === "array")) {
-    states.parentName = value;
+      const nodeId = addNodeToGraph({
+        graph,
+        text: String(nodeText),
+        type: child.type,
+      });
+      addEdgeToGraph(graph, parentId, nodeId);
+
+      // Call the appropriate traversal function
+      // Or end if there are no more nested elements
+      if (child.type === "array") {
+        traverseArray(states, array.children[i], nodeId);
+      } else if (child.type === "object") {
+        traverse({ objectToTraverse: child, states, myParentId: nodeId });
+      }
+    }
   }
-}
-
-function handleHasChildren(
-  type: NodeType,
-  states: States,
-  graph: Graph,
-  children: Node[],
-  myParentId?: string,
-  parentType?: string
-) {
-  let parentId: string | undefined;
-
-  if (type !== "property" && states.parentName !== "") {
-    // add last brothers node and add parent node
-
-    if (states.brothersNode.length > 0) {
-      const findBrothersNode = states.brothersNodeProps.find(
-        e =>
-          e.parentId === states.brothersParentId &&
-          e.objectsFromArrayId === states.objectsFromArray[states.objectsFromArray.length - 1]
-      );
-
-      if (findBrothersNode) {
-        const findNodeIndex = graph.nodes.findIndex(e => e.id === findBrothersNode?.id);
-
-        if (findNodeIndex !== -1) {
-          const modifyNodes = [...graph.nodes];
-          const foundNode = modifyNodes[findNodeIndex];
-
-          foundNode.text = foundNode.text.concat(states.brothersNode as any);
-          const { width, height } = calculateNodeSize(foundNode.text, false);
-
-          foundNode.width = width;
-          foundNode.height = height;
-
-          graph.nodes = modifyNodes;
-          states.brothersNode = [];
-        }
-      } else {
-        const brothersNodeId = addNodeToGraph({ graph, text: states.brothersNode });
-
-        states.brothersNode = [];
+};
 
-        if (states.brothersParentId) {
-          addEdgeToGraph(graph, states.brothersParentId, brothersNodeId);
+export const traverse = ({ objectToTraverse, states, myParentId }: Traverse) => {
+  // Unpack input arguments
+  const graph = states.graph;
+  const { children } = objectToTraverse;
+
+  // Setup initial step conditions
+  let nodeId = "";
+  const nodeText: [string, string][] = [];
+  const childQueue: Node[] = [];
+  if (children) {
+    // Loop over the children of the json node
+    for (let i = 0; i < children.length; i++) {
+      const child = children[i];
+      if (child.children) {
+        // If the child is not an object or array it is a property
+        // record it into the nodeText
+        // Otherwise, push it onto the nodes to be traversed
```

Comment:
```suggestion
        // If the child is not an object or array, it is a property;
        // record it into the nodeText.
        // Otherwise, push it onto the nodes to be traversed.
```